### PR TITLE
Change `excluded_territories.json` To `.toml`

### PR DESCRIPTION
**Pull request by AFg6K7h4fhy2**
For the scope of this PR, please refer to issue #148 .

### CODE REVIEW

**Comment by AFg6K7h4fhy2** [comment]
In `get_ensemble.R`, we should consider adding a command line argument for `--base_hub_path` to use here:

```
exclude_territories_path <- fs::path(
  base_hub_path,
  "auxiliary-data",
  "excluded_territories.toml"
)
if (fs::file_exists(exclude_territories_path)) {
  exclude_territories_toml <- RcppTOML::parseTOML(exclude_territories_path)
  excluded_locations <- exclude_territories_toml$locations
} else {
  stop("TOML file not found: ", exclude_territories_path)
}
```

**Comment by sbidari** [comment]
> In `get_ensemble.R`, we should consider adding a command line argument for `--base_hub_path` to use here:

Sounds good to me! After you add it, please also change it accordingly here https://github.com/CDCgov/covid19-forecast-hub/blob/d5a6f7dd7e2b9f1ec6824d23c0de44f46e9d4cb8/.github/workflows/create_ensemble.yaml#L42

**Comment by AFg6K7h4fhy2** [comment]
Not sure if this is correct (i.e. using `$(pwd)` for the directory): 

```
- name: generate ensemble
  run: |
    TODAY=$(date +'%Y-%m-%d')
    REF_DATE=$(date -d "$TODAY + 2 days" +'%Y-%m-%d') 
    Rscript src/code/get_ensemble.R --reference-date "$REF_DATE" --base_hub_path "$(pwd)"
```

Tagging DHM: @dylanhmorris

**Comment by sbidari** [comment]
> Not sure if this is correct (i.e. using `$(pwd)` for the directory):
> 
> ```
> - name: generate ensemble
>   run: |
>     TODAY=$(date +'%Y-%m-%d')
>     REF_DATE=$(date -d "$TODAY + 2 days" +'%Y-%m-%d') 
>     Rscript src/code/get_ensemble.R --reference-date "$REF_DATE" --base_hub_path "$(pwd)"
> ```
> 
> Tagging DHM: @dylanhmorris

GH Actions runs from the root of the repository, so you can just do "." . But `$(pwd)` should also work, do we need parenthesis around it though?

**Comment by dylanhmorris** [comment]
I think `.` is clearer. Notice that `src/code/get_ensemble.R` is a path too, so that same line of code already assumes our working dir is the repo root.

**Comment by AFg6K7h4fhy2** [comment]
>  so that same line of code already assumes our working dir is the repo root.

This is why I used `$(pwd)`. 

Re: <https://github.com/CDCgov/covid19-forecast-hub/pull/159#issuecomment-2536926312>

I will use "."

File: src/code/get_ensemble.R [hunk 1]
```diff
@@ -65,10 +71,17 @@ write.csv(
...
 )
-excluded_locations <- exclude_territory_data$locations
+if (fs::file_exists(exclude_territories_path)) {
+  exclude_territories_toml <- RcppTOML::parseTOML(exclude_territories_path)
+  excluded_locations <- exclude_territories_toml$locations
+} else {
+  stop("TOML file not found: ", exclude_territories_path)
+}
 current_forecasts <- current_forecasts |>
   dplyr::filter(model_id %in% models, !(location %in% excluded_locations))
```
Comment (sbidari) [inline src/code/get_ensemble.R:87]:
```suggestion
current_forecasts <- current_forecasts |>
  dplyr::filter(model_id %in% models)

```
I know I wrote this code and it's on me but just realized we don't need the locations exclusions here and the excluded territories bit above can be removed.

File: src/code/get_covid_hosp_data.R [hunk 2]
```diff
@@ -62,16 +62,17 @@ first_full_weekending_date <- args$first_full_weekending_date
...
+exclude_territories_path <- fs::path(
   base_hub_path,
   "auxiliary-data",
-  "excluded_territories.json"
+  "excluded_territories.toml"
 )
-if (!fs::file_exists(exclude_data_path)) {
-  stop("Exclude locations file not found: ", exclude_data_path)
+if (fs::file_exists(exclude_territories_path)) {
+  exclude_territories_toml <- RcppTOML::parseTOML(exclude_territories_path)
```
Comment (sbidari) [inline src/code/get_covid_hosp_data.R:71]: Could you also please add "RcppTOML" as dependency in `update-target-data.yaml`?

File: src/code/get_ensemble.R [hunk 3]
```diff
@@ -65,12 +71,19 @@ write.csv(
...
+  "auxiliary-data",
+  "excluded_territories.toml"
 )
-excluded_locations <- exclude_territory_data$locations
+if (fs::file_exists(exclude_territories_path)) {
+  exclude_territories_toml <- RcppTOML::parseTOML(exclude_territories_path)
+  excluded_locations <- exclude_territories_toml$locations
+} else {
+  stop("TOML file not found: ", exclude_territories_path)
+}
```
Comment (sbidari) [inline src/code/get_ensemble.R:84]: you can remove L73-84, I don't think these get used.

File: .github/workflows/create_ensemble.yaml [hunk 4]
```diff
@@ -31,15 +31,15 @@ jobs:
 
     - name: Install dependencies
       run: |
-        install.packages(c("hubEnsembles", "dplyr", "lubridate", "purrr", "yaml", "argparser", "remotes"))
+        install.packages(c("hubEnsembles", "dplyr", "lubridate", "purrr", "yaml", "argparser", "remotes", "RcppTOML"))
```
Comment (sbidari) [inline .github/workflows/create_ensemble.yaml:34]:
```suggestion
        install.packages(c("hubEnsembles", "dplyr", "lubridate", "purrr", "yaml", "argparser", "remotes"))
```

This is only required for `update-target-data.yaml` file here https://github.com/CDCgov/covid19-forecast-hub/blob/54627aeb384691e4eb1563eb7ebe687268b43ecc/.github/workflows/update-target-data.yaml#L33

**Review by sbidari — APPROVED** [review]
LGTM! Thanks @AFg6K7h4fhy2

**Pull request merged**